ibis/backends/base/sql/registry/binary_infix.py

```diff
@@ -50,37 +48,41 @@ def xor(translator, op):
     return "({0} OR {1}) AND NOT ({0} AND {1})".format(left_arg, right_arg)
 
 
-def contains(op_string: Literal["IN", "NOT IN"]) -> str:
-    def translate(translator, op):
-        from ibis.backends.base.sql.registry.main import table_array_view
-
-        if isinstance(op.options, tuple) and not op.options:
-            return {"NOT IN": "TRUE", "IN": "FALSE"}[op_string]
-
-        left = translator.translate(op.value)
-        if helpers.needs_parens(op.value):
-            left = helpers.parenthesize(left)
-
-        ctx = translator.context
-
-        if isinstance(op.options, tuple):
-            values = [translator.translate(x) for x in op.options]
-            right = helpers.parenthesize(", ".join(values))
-        elif op.options.shape.is_columnar():
-            right = translator.translate(op.options)
-            if not any(
-                ctx.is_foreign_expr(leaf)
-                for leaf in an.find_immediate_parent_tables(op.options)
-            ):
-                array = op.options.to_expr().as_table().to_array().op()
-                right = table_array_view(translator, array)
-            else:
-                right = translator.translate(op.options)
-        else:
-            right = translator.translate(op.options)
-
-        # we explicitly do NOT parenthesize the right side because it doesn't
-        # make sense to do so for Sequence operations
-        return f"{left} {op_string} {right}"
-
-    return translate
+def in_values(translator, op):
+    if not op.options:
+        return "FALSE"
```

Comment:
This seems like a really strange edge, but such is life 😮‍💨

Reply:
Well, we could avoid it on the expression level, but that can be done in a follow-up.